src/plan/semispace/global.rs

```diff
@@ -141,12 +142,8 @@ impl<VM: VMBinding> Plan for SemiSpace<VM> {
         &self.common
     }
 
-    fn global_side_metadata_per_chunk(&self) -> usize {
-        if !VM::VMObjectModel::HAS_GC_BYTE {
-            meta_bytes_per_chunk(3, 1)
-        } else {
-            0
-        }
+    fn global_side_metadata_spec_vec(&self) -> &Vec<SideMetadataSpec> {
+        &self.metadata_spec_vec
```

Comment:
`SideMetadataSpec` constants are only essential in the performance-critical functions such as load/store.
For those cases, we always pass a single object (constant instance of `SideMetadataSpec`).
By the way, I'll change the implementation to what you proposed, mainly because I think it's cleaner